content/docs/bdd/better-gherkin.md

```diff
@@ -35,3 +35,47 @@ In fact, a good question to ask yourself when writing a feature clause is: “Wi
+
+Imperative tests are communicative, but because they are so closely tied to the mechanics of the current UI, they are often more work to maintain. It's easiest to understand this by looking at examples.
+
+Here's a more imperative style;
+```
+Feature: Subscribers see different sets of stock images based on their subscription level 
+
+Scenario: Free subscribers see only the free articles
+
+  Given users with a free subscription can access "FreeArticle1" but not "PaidArticle1" 
+  When I type "freeFrieda@example.com" in the email field
+  And I type "validPassword123" in the password field
+  And I press the "Submit" button
+  Then I see "FreeArticle1" on the home page
+  And I do not see "PaidArticle1" on the home page
+
+Scenario: Subscriber with a paid subscription can access "FreeArticle1" and "PaidArticle1"
+  Given I am on the login page
+  When I type "paidPattya@example.com" in the email field
+  And I type "validPassword123" in the password field
+  And I press the "Submit" button
+  Then I see "FreeArticle1" and "PaidArticle1" on the home page  
+```
+
+Each step is a precise instruction. It is clear what is done, and the inputs and expected results are specified exactly. But it's easy to imagine changes to the application which would require changing these tests. The available options for free versus paid subscriptions can change. Even the means of logging in could change. What if, in the future, users log in with a voice interface or a thumbprint? 
+
+A more declarative style hides the details of how the application's capabilities are implemented.
+
+```
+  Given Free Frieda has a free subscription
+  When Free Frieda logs in with her valid credentials
+  Then she sees a Free article on the home page
+
+  Given Paid Patty has a basic-level paid subscription
+  When Paid Patty logs in with her valid credentials
+  Then she sees a Free article and a Paid article on the home page
+``` 
+With a declarative style, each step communicates an idea, but the exact values aren't specified. The details, such as what specific articles are free or paid, and the subscription level of different test users, are specified "behind the scenes" in the step definition code. The subscription packages can change in the future. The business can change what content is available to subscribers on free and paid plans, without having to change this test and other tests that use the same step definitions. If another subscription level is added later, it's easy to add a test for that. By avoiding terms like “click a button” that suggest implementation, the test is more resilient to implementation details of the UI. The intent of the test remains the same, even if the implementation changes later.
+
+Imperative tests do have a place, but use them wisely. There’s no right or wrong, experiment to see what fits your context.  
```

Comment:
Not sure about this paragraph. The rest of the content is all about being more declarative, so this feels out of place to me. Consider removing this section?

Reply:
I did remove it, but I added a bit on line 43 trying to convey that it is not *wrong* to use imperative style, it can be the best choice for some contexts. Is that ok?

Reply:
Yes, it's much clearer now, thank you!